airflow/contrib/hooks/azure_hdinsight_hook.py

```diff
@@ -0,0 +1,252 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+import json
+import logging
+import time
+
+import requests
+import urllib3
+from azure.common.credentials import ServicePrincipalCredentials
+from azure.mgmt.hdinsight import HDInsightManagementClient
+from azure.mgmt.resource import ResourceManagementClient
+from azure.mgmt.resource.resources.models import DeploymentMode, DeploymentProperties
+
+from airflow.exceptions import AirflowException
+from airflow.hooks.base_hook import BaseHook
+from airflow.utils.log.logging_mixin import LoggingMixin
+
+urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+
+
+class AzureHDInsightHook(BaseHook, LoggingMixin):
+
+    def __init__(self, azure_conn_id='azure_hdinsight_default'):
```

Comment:
Is azure_default not valid?

Reply:
actually you're right, i want to use ```azure_default  ``` but  Azure uses Hortonwork(HDP) VM  and each HDP instance  have their authorization (username , password and they  are being generated at creating cluster by user) independent from  Azure Creddential . So 'm confused ?

Reply:
Otherwise. Integration into the Azure platform is at an early stage of development. There are no good practices yet. I ask a lot of questions to force a discussion that can create these practices.
Could you review other unacceptable patches? I am happy to hear your opinion.
https://github.com/apache/airflow/pulls?q=is%3Apr+Azure+is%3Aopen

Reply:
Good practice for GCP are available: https://docs.google.com/document/d/1_rTdJSLCt0eyrAylmmgYc3yZr-_h51fVlnvMmWqhCkY/edit
However, the document is still developing all the time.